cirq-ft/cirq_ft/algos/phase_estimation.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2023 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Optional, List
+from attr import frozen
+from cirq._compat import cached_property
+import cirq
+
+from cirq_ft import infra
+
+
+@frozen
+class KitaevPhaseEstimation(infra.GateWithRegisters):
+    r"""Class representing the Kitaev Phase Estimation algorithm, originally introduced by
+    Kitaev in https://arxiv.org/abs/quant-ph/9511026."""
+
+    precision: int
+    U: cirq.Gate
+    eigenvector_prep: Optional[infra.GateWithRegisters] = None
+
+    @cached_property
+    def registers(self) -> infra.Registers:
+        return infra.Registers.build(
+            bits_of_precision_register=self.precision, eigenvector_register=self.eigenvector_bitsize
+        )
+
+    @cached_property
+    def eigenvector_bitsize(self):
+        return self.U.num_qubits()
+
+    def qft_inverse(self, qubits):
+        """Generator for the inverse QFT on a list of qubits."""
+        qreg = list(qubits)
+        while len(qreg) > 0:
+            q_head = qreg.pop(0)
+            yield cirq.H(q_head)
+            for i, qubit in enumerate(qreg):
+                yield (cirq.CZ ** (-1 / 2 ** (i + 1)))(qubit, q_head)
```

Comment:
I figured out the issue with `cirq.qft` -- The output it gives you is reversed and the `without_reverse=True` doesn't work as expected when `inverse=True` is set. To fix it, we can simply insert another layer of swaps and then your tests will pass -- you can see my implementation for more details. 

For the sake of this PR, I'm fine if you want to use your own implementation of QFT inverse but maybe open an issue to track the improvements to `cirq.qft` so we can use it here once it fixed. Even better, consider doing these improvements if you have some spare cycles!